src/validators/validators.ts

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable @typescript-eslint/no-namespace */
+
+import { Validator } from "../core";
+import { Primitive, WidenType } from "../utils/utility-types";
+
+import * as BaseErrors from "./base-errors";
+
+// utils
+
+/**
+ * Creates new validator that is a wrapper around provided validator
+ * but returning different errors
+ *
+ * @param validator used to construct new validator
+ * @param error errors returned by the new validator
+ */
+export const withError = <T, Err1, Err2>(
+  validator: Validator.Sync<T, Err1>,
+  error: Err2
+): Validator.Sync<T, Err2> => val => (validator(val) == null ? null : error);
+
+// general
+
+/** Checks if field value is present */
+export const required = (): Validator.Sync<
+  unknown,
+  BaseErrors.Required
+> => val =>
+  val == null || val === "" || val === false ? { code: "required" } : null;
+
+/**
+ * Special validator that breaks validation chains early with no error, when field value is not present.
+ * This means that other validators do not have to handle empty value as edge case.
+ */
+export const optional = () => (val: unknown): null => {
+  const notPresent = required()(val) != null;
+
+  if (notPresent) {
+    throw true; // special way of breaking validation chain early
```

Comment:
elegant hack, but what do you think about allowing validators to return some reserved value to break the validation chain early?
not sure if this feature will be useful, but this case proves it could be, and imho it's not that obvious that `throw true` will do the trick (even if documented)
smth like that?
```
if(notPresent) {
  return "break-validation-chain"
}
```

Reply:
I don't think it matters if it's for internal use only. if it needs to be for use by consumers as well then maybe exported Symbol would be good. 

I'd leave it for now as internal thing, and if we want to add it as part of public API then we will improve it